coreos-installer

```diff
@@ -378,6 +378,42 @@ write_networking_opts() {
     echo "set ignition_network_kcmdline=\"$(cat /tmp/networking_opts)\"" >> /mnt/boot_partition/ignition.firstboot
 }
 
+#########################################################
+# Helper to persist ignition platform id to initramfs first boot
+#########################################################
+write_platform_id() {
+    local tmp_platform_id="/tmp/platform_id"
+    local proc_cmdline="/proc/cmdline"
+
+    if [ ! -f ${tmp_platform_id} ]; then
+        log "Not overwriting ignition platform id, no platform id provided"
+        return
+    else
+        local platform_id="$(cat ${tmp_platform_id})"
+    fi
+
+
+    log "Overwriting ignition platform id"
+    # check for the boot partition
+    mkdir -p /mnt/boot_partition
+    mount_boot_partition /mnt/boot_partition
+    trap 'umount /mnt/boot_partition; trap - RETURN' RETURN
+    local conf_files=$(ls /mnt/boot_partition/loader/entries/*conf)
+
+    for f in $conf_files
+    do
+        # Update boot loader entry first
+        if grep -q 'ignition.platform.id' $f; then
+            sed -i "s/ignition.platform.id=.* /ignition.platform.id=${platform_id} /" $f
```

Comment:
`.*` is greedy, so this will clobber any following kargs.  It will also fail if `ignition.platform.id` is the last arg but isn't followed by a space.

Reply:
So maybe `[a-z]*`? For EOL case, what about `[a-z]*$` to match the case?

Reply:
A regex that matches anything but space and EOL should work. Here is an example: 

```
[dustymabe@media tmp]$ cat bls.conf 
options root=/dev/disk/by-label/root rootflags=defaults,prjquota rw $ignition_firstboot mitigations=auto,nosmt console=tty0 console=ttyS0,115200n8 ignition.platform.id=qemu ostree=/ostree/boot.1/fedora-coreos/030433db74a4e975020ed5c806c9e4e066665480e05f43463024bfde3b90e02e/0
options root=/dev/disk/by-label/root rootflags=defaults,prjquota rw $ignition_firstboot mitigations=auto,nosmt console=tty0 console=ttyS0,115200n8 ostree=/ostree/boot.1/fedora-coreos/030433db74a4e975020ed5c806c9e4e066665480e05f43463024bfde3b90e02e/0 ignition.platform.id=qemu
[dustymabe@media tmp]$ 
[dustymabe@media tmp]$ sed "s/ignition.platform.id=[^$ ]*/ignition.platform.id=foo/" bls.conf
options root=/dev/disk/by-label/root rootflags=defaults,prjquota rw $ignition_firstboot mitigations=auto,nosmt console=tty0 console=ttyS0,115200n8 ignition.platform.id=foo ostree=/ostree/boot.1/fedora-coreos/030433db74a4e975020ed5c806c9e4e066665480e05f43463024bfde3b90e02e/0
options root=/dev/disk/by-label/root rootflags=defaults,prjquota rw $ignition_firstboot mitigations=auto,nosmt console=tty0 console=ttyS0,115200n8 ostree=/ostree/boot.1/fedora-coreos/030433db74a4e975020ed5c806c9e4e066665480e05f43463024bfde3b90e02e/0 ignition.platform.id=foo

```

The `[^$ ]*` says match anything that is not (`^`) EOL (`$`) or space (` `) and do this any number of times (`*`).

Reply:
@dustymabe That looks better :+1:

Reply:
Updated.

Reply:
Might also be worthwhile to switch using a PCRE regex via the `-r` flag and use `[^$ ]*?` to do a reluctant match rather than a greedy match.

Reply:
@arithx Updated :+1: